Stellar-contract.x

```diff
@@ -270,27 +226,81 @@ case SC_ADDRESS_TYPE_CONTRACT:
     Hash contractId;
 };
 
-union SCObject switch (SCObjectType type)
+%struct SCVal;
+%struct SCMapEntry;
+
+const SCVAL_LIMIT = 256000;
+const SCSYMBOL_LIMIT = 256;
+
+typedef SCVal SCVec<SCVAL_LIMIT>;
+typedef SCMapEntry SCMap<SCVAL_LIMIT>;
+
+typedef string SCString<SCVAL_LIMIT>;
+typedef string SCSymbol<SCSYMBOL_LIMIT>;
+
+union SCVal switch (SCValType type)
 {
-case SCO_VEC:
-    SCVec vec;
-case SCO_MAP:
-    SCMap map;
-case SCO_U64:
+
+case SCV_BOOL:
+    bool b;
+case SCV_VOID:
+    void;
+case SCV_STATUS:
+    SCStatus error;
+
+case SCV_U32:
+    uint32 u32;
+case SCV_I32:
+    int32 i32;
+
+case SCV_U64:
     uint64 u64;
-case SCO_I64:
+case SCV_I64:
     int64 i64;
-case SCO_U128:
+case SCV_TIMEPOINT:
+    uint64 timepoint;
```

Comment:
Can we use the timepoint type if the goal is to align this value with that existing concept in the XDR?
```suggestion
    Timepoint timepoint;
```

Reply:
Fixed (will push a rebased change)